enhancements/security/x509-trust.md

```diff
@@ -0,0 +1,213 @@
+The default trust for containers like the machine-API operator is that container's CA store, which depends on the base layer ([Red Hat Universal Base Image (UBI)][ubi], etc.).
+Many external components will have X.509 certificates signed by one of those default CAs, but this is not always the case.
+
+There may also be internal communication between cluster components that relies on X.509 trust but lacks a custom 
+The framework from this enhancement can also be used to configure trust for that communication as well.
+
+This enhancement provides a mechanism for configuring additional or alternative trust bundles for all cluster services that do not already have existing trust configuration mechanisms.
+
+## Motivation
+
+### Goals
+
+Administrators will have a single location to configure a default additional trust bundle to be used by all cluster components.
+They will also be able to specify whether this default is to be use in addition to, or instead of, the component's default trust bundle.
+
+Administrators will have per-component configuration locations to adjust the cluster-scoped default.
+As for the cluster-scoped default, they will be able to specify whether the per-component trust bundle is to be used in addition to, or instead of, the cluster-scoped default.
+
+### Non-Goals
+
+This enhancement does not address components making intra-cluster TLS requests.
+Trust for those requests is covered by [the Kubernetes API-server Certificates enhancement](../kube-apiserver/certificates) and [the Automated Service CA Rotation enhancement](../automated-service-ca-rotation.md).
+
+This enhancement does not address configuration for non-CA TLS settings like ciphers and allowed TLS versions.
+[The ingress TLS configuration enhancement](../kube-apiserver/tls-config.md) covers those settings for cluster-hosted services.
+Outgoing TLS probably needs similar configuration, but that configuration is decoupled from X.509 trust and should be addressed in a separate enhancement.
+
+## Proposal
+
+### Cluster-scoped Default Trust
+
+There will be a ConfigMap (FIXME: or should this be a new CRD under openshift/api?) named `default-ca-bundle` in the `openshift-config` namespace.
+The ConfigMap will have the following keys:
+
+* `ca-bundle.crt`, containing a PEM-encoded X.509 certificate bundle.
+* `composition`, configuring how the bundle in `ca-bundle.crt` is composed with the component's default trust store.
+    Valid values are:
+    * `union`, in which case components should use the union of the `ca-bundle.crt` and their default trust store.
+    * `override`, in which case components should use only `ca-bundle.crt` and ignore their default trust store.
+        This behavior is also the default when a component recieves an unrecognized `composition` value.
+
+### Per-component Trust
```

Comment:
I don’t see a justification for this.  We already have image additionalTrustedCA - perhaps that field is the bug, and we should have a single cluster wide trust bundle, but why are components defining their own trust?  Who really needs that?

Reply:
> Who really needs that?

Answered [over here][1].

[1]: https://github.com/openshift/enhancements/pull/115#discussion_r349958510